bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsAccountHandler.java

```diff
@@ -0,0 +1,332 @@
+            localPollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    private void poll() {
+        logger.debug("Polling");
+
+        // when access token not yet received or expired, login and get devices in next refresh cycle
+        if (accessToken == null) {
+            login();
+            return;
+        }
+
+        getControllers().handle((devicesList, accountException) -> {
+            if (devicesList != null) {
+                logger.trace("deviceList response: {}", devicesList);
+
+                List<ElroConnectsConnector> response = null;
+                try {
+                    response = gson.fromJson(devicesList, deviceListType);
+                } catch (JsonParseException parseException) {
+                    logger.warn("Parsing failed: {}", parseException.getMessage());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "@text/offline.request-failed");
+                    return null;
+                }
+                Map<String, ElroConnectsConnector> devices = new HashMap<>();
+                if (response != null) {
+                    response.forEach(d -> devices.put(d.getDevTid(), d));
+                }
+                this.devices = devices;
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                if (accountException == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            accountException.getLocalizedMessage());
+                }
+            }
+
+            return null;
+        });
+    }
+
+    private void login() {
+        logger.debug("Login");
+        getAccessToken().handle((accessToken, accountException) -> {
+            this.accessToken = accessToken;
+            if (accessToken != null) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                if (accountException == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            accountException.getLocalizedMessage());
+                }
+            }
+
+            return null;
+        });
+    }
+
+    private CompletableFuture<@Nullable String> getAccessToken() {
+        CompletableFuture<@Nullable String> f = new CompletableFuture<>();
+
+        // The method call returns an invalid 401 response on authentication error, missing the www-authentication
+        // header. This header should be there according to RFC7235. This workaround removes the protocol handler and
+        // the check.
+        client.getProtocolHandlers().remove(WWWAuthenticationProtocolHandler.NAME);
```

Comment:
@jlaur Is this acceptable?

Reply:
I don't have an opinion about this or ideas how to approach it differently. If this is needed for the desired behavior, I'm fine with it.

Reply:
@jlaur After reconsideration, what I did is not a good idea. This will change the behaviour of the common http client and disable all BASIC and DIGEST automatic authentication handling on the common http client. By consequence other bindings using this automatic handling will fail.
I will change it, creating a dedicated http client in the handler factory.